index.js

```diff
@@ -0,0 +1,166 @@
+//Napisz program, który będzie symulować działanie pralki (takiej do prania ubrań)
+//Twoja pralka powinna umożliwiać użytkownikowi wybranie programu prania (szybkie, długie)
+//Powinna umieć wirować (na koniec szybkiego prania, oraz na koniec i w połowie długiego)
+//Po wybraniu jakiegokolwiek programu powinna odpompować starą wodę, pobrać nową.
+//Na podstawie wybranego programu podgrzać każdorazowo wodę do odpowiedniej temperatury.
+//I odpompować ją przed wirowaniem.Powinna umożliwiać użytkownikowi uruchomić wirowanie na żądanie.
+//Aby zabezpieczyć łazienkę przed zalaniem, pralka powinna ryglować drzwi pralki po włączeniu
+//jakiegokolwiek programu oraz odblokowywać 30 sekund po skończonym praniu.
+//Pralka powinna płukać ubrania przed wirowaniem, oraz udostępniać płukanie z wirowaniem jako
+//osobny program do wybrania.
+//Każda pralka bierze proszek, więc programy piorące powinny sprawdzać czy użytkownik
+//wsypał proszek do prania.
+//Pralka powinna sprawdzać czy użytkownik włożył ciuchy do bębna.
+// W przypadku ich braku powinna poinformować użytkownika o tym.
+
+class WashingmMachine {
+  constructor() {
+    this._program = null;
+    this._quickProgram = false;
+    this._longProgram = false;
+    this._spin = false;
+    this._drain = false;
+    this._locked = true;
+    this._rinse = false;
+    this._rinseAndSpin = false;
+    this._detergentAdded = false;
+    this._clothesAdded = false;
+  }
+
+  addDetergent = () => {
+    this._detergentAdded = true;
+    console.log("Dodano proszek do pralki");
+  };
+
+  addClothes = () => {
+    this._clothesAdded = true;
+    console.log("Włożyłeś ubrania do pralki");
+  };
+
+  checkDetergent = () => {
+    if (!this._derergentAdded) {
+      console.log("Nie dodano proszku - najpierw wsyp proszek");
+    }
+  };
+  checkClothes = () => {
+    if (!this._clothesAdded) {
+      console.log("Nie włożyłeś ubrań - włóż najpierw ubrania");
+    }
+  };
+
+  chooseProgram = (program) => {
+    if (!this.checkClothes()) {
+      return;
+    }
+    if (!this.checkDetergent()) {
+      return;
+    }
+    this._program = program;
+    if (program === "szybkie") {
```

Comment:
choose program powinien tylko tworzyc nowa instancje klasy program i przypisywac ją do this._program